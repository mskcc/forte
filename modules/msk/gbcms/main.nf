process GBCMS {
    tag "$meta.id"
    label 'process_single'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'ghcr.io/msk-access/gbcms:1.2.5':
        'ghcr.io/msk-access/gbcms:1.2.5' }"
    
    input:
    tuple val(meta), path(bam), path(bambai), path(variant_file), val(output)
    path(fasta) 
    path(fastafai)

    output:
     tuple val(meta), path('*.{vcf,maf}') , emit: variant_file
     tuple val(meta), path("versions.yml")   , emit: versions

    when:
        task.ext.when == null || task.ext.when
    script:
    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        error "GetBaseCountsMultiSample module does not support Conda. Please use Docker / Singularity instead."
    }
    def args = task.ext.args ?: ''
    def sample = meta.sample
    // determine if input file is a maf of vcf 

    def input_ext = variant_file.getExtension()
    def variant_input = ''
    
    if(input_ext == 'maf') {
        variant_input = '--maf ' + variant_file
    } 
    if(input_ext == 'vcf'){
            variant_input = '--vcf ' + variant_file
    }
    if(variant_input == ''){
        throw new Exception("Variant file must be maf or vcf, not ${input_ext}")
    }

    """
    GetBaseCountsMultiSample --fasta ${fasta} \\
    ${variant_input} \\
    --output ${output} \\
    --bam $sample:${bam} $args
    
    cat <<-END_VERSIONS > versions.yml
    "${task.process}": 
        GetBaseCountsMultiSample: \$(echo \$(GetBaseCountsMultiSample --help) | grep -oP '[0-9]\\.[0-9]\\.[0-9]')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"

    """

    touch variant_file.maf
    cat <<-END_VERSIONS > versions.yml
    "${task.process}": 
        GetBaseCountsMultiSample:  1.2.5
    END_VERSIONS
    """
}
