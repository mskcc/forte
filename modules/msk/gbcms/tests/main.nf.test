// nf-core modules test gbcms
nextflow_process {

    name "Test Process GBCMS"
    script "../main.nf"
    process "GBCMS"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "gbcms"

    test("sarscov2 illumina  paired-end [fastq]") {

        when {
            process {
                """
                input[0] = [
                            [ id:'test', sample:'197' ], // meta map
                            file(params.test_data['sarscov2']['illumina']['test_single_end_sorted_bam'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_single_end_sorted_bam_bai'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_vcf'], checkIfExists: true),
                            "variant_file.vcf" 
                        ]
                input[1] = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                input[2] = file(params.test_data['sarscov2']['genome']['genome_fasta_fai'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
