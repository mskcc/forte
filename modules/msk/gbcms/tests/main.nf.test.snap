{
    "sarscov2 illumina  paired-end [fastq]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "sample": "197"
                        },
                        "variant_file.vcf:md5,28c8df33c7ea5ed5d1cf9997d8e00ffa"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "sample": "197"
                        },
                        "versions.yml:md5,0ca451c1c6505a0a72541b6cc29e2bc0"
                    ]
                ],
                "variant_file": [
                    [
                        {
                            "id": "test",
                            "sample": "197"
                        },
                        "variant_file.vcf:md5,28c8df33c7ea5ed5d1cf9997d8e00ffa"
                    ]
                ],
                "versions": [
                    [
                        {
                            "id": "test",
                            "sample": "197"
                        },
                        "versions.yml:md5,0ca451c1c6505a0a72541b6cc29e2bc0"
                    ]
                ]
            }
        ],
        "timestamp": "2023-12-13T15:05:34.981489"
    }
}